src/main/java/seedu/duke/Cache.java

```diff
@@ -0,0 +1,94 @@
+package seedu.duke;
+
+import seedu.duke.ui.Ui;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Cache {
+    private static final String CACHE_FILE_PATH = "./data/player_list.txt";
+    private static final String CACHE_FILE_DIR = "./data";
+    private static final String CACHE_FILE_DELIM = " \\| ";
+    //stores the previous game state for each player
+    public static ArrayList<Player> players = new ArrayList<>();
+
+    /**
+     * Creates a file called "player_list.txt" to cache user data in directory "data"
+     * Additionally, creates a new directory called "data" if not created yet
+     */
+    public static void spawnCacheFile() {
+        File f = new File(CACHE_FILE_PATH);
+        boolean isSpawnSuccess = false;
+        while (!isSpawnSuccess) {
+            try {
+                isSpawnSuccess = f.createNewFile();
+            } catch (IOException e) {
+                new File(CACHE_FILE_DIR).mkdirs();
+            }
+        }
+    }
+
+    /**
+     * Reads user data from cache file and loads all user accounts based on read data
+     *
+     * @throws IllegalArgumentException If zone is <= 0.
+     */
+    public static void loadPlayers() throws FileNotFoundException {
```

Comment:
Consider changing the name to loadData? As I think we may confuse loadPlayers with the load command.